File: backend/app/schemas/onboarding.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class SelectMethodRequest(BaseModel):
    method: str  # "integration" | "documents" | "manual_customers"


class OnboardingStep(BaseModel):
    key: str
    label: str
    completed: bool


class OnboardingStatusOut(BaseModel):
    onboarding_method: str | None
    steps: list[OnboardingStep]
    first_insight_at: datetime | None


class FirstInsightOut(BaseModel):
    type: str  # "risk_flag" | "all_clear"
    message: str
    ai_action_id: UUID | None


class OnboardingCompleteOut(BaseModel):
    onboarding_method: str | None
    steps: list[OnboardingStep]
    first_insight_at: datetime | None
    first_insight: FirstInsightOut | None

File: frontend/lib/onboarding.ts
import { apiFetch } from "@/lib/api-client";

export type OnboardingStep = { key: string; label: string; completed: boolean };

export type OnboardingStatus = {
  onboarding_method: string | null;
  steps: OnboardingStep[];
  first_insight_at: string | null;
};

export type FirstInsight = {
  type: "risk_flag" | "all_clear";
  message: string;
  ai_action_id: string | null;
};

export type OnboardingCompleteResult = OnboardingStatus & {
  first_insight: FirstInsight | null;
};

export function getOnboardingStatus(token: string): Promise<OnboardingStatus> {
  return apiFetch<OnboardingStatus>("/onboarding/status", { token });
}

export function selectOnboardingMethod(
  method: "integration" | "documents" | "manual_customers",
  token: string,
): Promise<OnboardingStatus> {
  return apiFetch<OnboardingStatus>("/onboarding/select-method", {
    method: "POST",
    body: { method },
    token,
  });
}

export function completeOnboarding(token: string): Promise<OnboardingCompleteResult> {
  return apiFetch<OnboardingCompleteResult>("/onboarding/complete", { method: "POST", token });
}
